src/app/api/utils/auth.ts

```diff
@@ -92,8 +85,10 @@ export const authOptions: AuthOptions = {
   ],
   callbacks: {
     // Unused arguments also listed to show what's available:
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     async jwt({ token, account, profile, trigger }) {
+      if (trigger === "update") {
+        profile = await fetchUserInfo(token.subscriber?.fxa_access_token ?? "");
```

Comment:
I think you should also pass this through the `authOptions.providers[0].profile` callback to ensure that the format is consistent. (Possibly also extracting that out into a separate function.)

Reply:
Hmm that's really weird, I was pretty sure (see also the comment about the `profile` function) that it parsed the return value. The primary difference I think is that we use the `uid` from the FxA profile as the `id` used by `Next-Auth`. If the object returned by `/userinfo/` does not include a `.uid` property, then you're probably right and we can merge this.

Reply:
Oh OK yes stepping through the output in the debugger you're right... `/userinfo/` does return an `FxaProfile`. OK I see your point then, I'm not sure why that passed typechecking - I suspect it's just because the `jwt` callback doesn't make use of any of these extra/conflicting fields like `uid` vs. `id` an `amrValues` - it just wants to create a token like:

```
        token.fxa = {
          locale: profile.locale,
          twoFactorAuthentication: profile.twoFactorAuthentication,
          metricsEnabled: profile.metricsEnabled,
          avatar: profile.avatar,
          avatarDefault: profile.avatarDefault,
          subscriptions: profile.subscriptions,
        };
```

I'll go ahead and do what you originally suggested.

Reply:
That is, we should have a common way to take an `FxaProfile` and convert that to a standard `Profile`.